src/coreclr/gc/gc.cpp

```diff
@@ -24910,6 +24929,8 @@ void gc_heap::recommission_heap()
 
 void gc_heap::check_heap_count ()
 {
+    dynamic_heap_count_data.new_n_heaps = n_heaps;
```

Comment:
since you are doing this here, you don't need to do it again here since it doesn't change inbetween

```cplusplus
    if (gc_heap::background_running_p())
    {
        // can't have background gc running while we change the number of heaps
        // so it's useless to compute a new number of heaps here
        dynamic_heap_count_data.new_n_heaps = n_heaps;
    }
    else
```

Reply:
You're right. Thanks for spotting!